### ISSUE TITLE
Custom layer does not show up in layer list

### ISSUE THREAD

**Issue by patmjen** [napari/napari#2757]
## 🐛 Bug

I'm trying to create some custom layers that add functionality to existing napari layers. When I add them to the viewer, they don't show up in the layer list. But, they have been added as the layer controls are there, and I can also select, hide, etc. in the layer list. They are just not shown.

## To Reproduce

Running the following code shows the bug for me:
```python
import numpy as np
import napari

class NewLabels(napari.layers.Labels):
    """'Empty' extension of napari Labels layer."""
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)

# Make a viewer and add the custom layer
viewer = napari.Viewer()
viewer.add_layer(NewLabels(np.zeros((10, 10, 10), dtype=np.uint8)))
```

## Expected behavior
I'd expect the `NewLabels` layer to show up in the layer list as a Labels layer.

## Environment

I'm using Python 3.7.4 on Windows 10 and napari 0.4.8. I'm running in a jupyter notebook.

## Additional context
In the console where I started jupyter lab, I get the following error message when I try to add the layer:
```
Traceback (most recent call last):
  File "D:\Anaconda3\lib\site-packages\napari\_qt\containers\_layer_delegate.py", line 81, in paint
    self.get_layer_icon(option, index)
  File "D:\Anaconda3\lib\site-packages\napari\_qt\containers\_layer_delegate.py", line 96, in get_layer_icon
    icon = QColoredSVGIcon.from_resources(icon_name)
  File "D:\Anaconda3\lib\site-packages\napari\_qt\qt_resources\_svg.py", line 110, in from_resources
    path = get_icon_path(icon_name)
  File "D:\Anaconda3\lib\site-packages\napari\resources\_icons.py", line 20, in get_icon_path
    icons=set(ICONS),
ValueError: unrecognized icon name: 'new_newlabels'. Known names: {'fill', 'chevron_down', 'polygon', 'check', 'paint', 'long_right_arrow', 'delete_shape', 'rectangle', 'step_right', 'new_points', 'add', 'pop_out', 'chevron_up', 'new_surface', 'minus', 'logo_silhouette', 'copy_to_clipboard', 'left_arrow', 'visibility', 'home', 'direct', 'zoom', 'chevron_left', 'new_image', 'select', 'shuffle', '3D', 'plus', 'long_left_arrow', 'delete', 'picker', 'move_front', 'transpose', 'roll', 'down_arrow', 'right_arrow', 'step_left', 'move_back', 'vertex_insert', 'new_vectors', 'ellipse', 'circle', 'square', 'console', '2D', 'visibility_off', 'warning', 'line', 'info', 'path', 'vertex_remove', 'drop_down', 'new_labels', 'up_arrow', 'new_tracks', 'new_shapes', 'grid', 'erase'}
``´

**Comment by tlambert03** [comment]
thanks!  will get on it asap

**Comment by tlambert03** [comment]
this particular bug is fixed in #2758... but we don't do a lot of testing of custom layer types, so please let us know if you run into any other issues!
